juju/juju
# Update to use the new txn.Runner that tracks Duration.

## Description of change

Since txn.Runner now makes Clock mandatory, we have to update all the
callers to pass in a clock.
This can't land as is, because gopkg.in/blobstore.v2 also needs the
updated juju/txn and pass in the Clock there as well.

But this can land as soon as we land the existing patch against
blobstore.
https://github.com/juju/blobstore/pull/28

## QA steps

```
$ juju model-config -m controller logging-config="<root>=INFO;juju.state.txn=TRACE"
$ juju debug-log -m controller
```

Should allow you to see the transactions being run, and how long each one is taking. eg:
```
machine-0: 09:28:06 TRACE juju.state.txn ran transaction in 0.040s []txn.Op{
    {
        C:      "leases",
        Id:     "ee39ccec-e072-44e2-85e2-7ca5cd8d46f3:application-leadership#ul9#",
        Assert: bson.M{
            "holder":   "ul9/0",
            "start":    int64(192061919440),
            "duration": time.Duration(60000000000),
            "writer":   "machine-0",
        },
        Insert: nil,
        Update: bson.M{
            "$set": bson.D{
                {
                    Name:  "start",
                    Value: int64(221070101461),
                },
                {
                    Name:  "duration",
                    Value: int64(60000000000),
                },
                {
                    Name:  "writer",
                    Value: "machine-0",
                },
            },
        },
        Remove: false,
    },
}
err: <nil>
```

## Documentation changes

We could explicitly document the above logging config to expose transactions.

## Bug reference

None.
